### DISCUSSION TITLE
Calling functions from JS yeilds TypeErrors

### DISCUSSION THREAD

**Discussion by GageHowe** [wasm-bindgen/wasm-bindgen#3833] [Q&A]
### Summary

When calling a rust->wasm function from main.js, the inspector shows an error: "TypeError: Cannot read properties of undefined (reading '__wbindgen_add_to_stack_pointer')"
Any thoughts? I've been debugging for hours to no avail.

### Additional Details

https://github.com/GageHoweTamu/PWGen
A chromium browser extension

**Comment by Liamolucko** [comment]
It looks like you're getting that error because you aren't running `await init()` before trying to use the Wasm module.

**Reply by GageHowe** [reply]
Thanks for the help. As a wasm-bindgen newbie I tried that but got another error and the extension stopped working:

Uncaught CompileError: WebAssembly.instantiateStreaming(): Refused to compile or instantiate WebAssembly module because neither 'wasm-eval' nor 'unsafe-eval' is an allowed source of script in the following Content Security Policy directive: "script-src 'self'" 

Do you happen to know how to resolve this? I added:
<meta http-equiv="Content-Security-Policy" content="script-src 'self' 'wasm-eval';">
to the index.js.

I greatly appreciate the help.

EDIT: turns out Chrome's content security policy doesn't play nicely with WASM, which explains the weirdness. I ended up rewriting it in JS
